gateway/api/use_cases/get_provider_jobs.py

```diff
@@ -0,0 +1,79 @@
+"""This module contains the usecase get_jobs"""
+import logging
+from datetime import datetime
+from typing import List, Tuple, Optional
+from api.models import Job, VIEW_PROGRAM_PERMISSION
+from api.access_policies.providers import ProviderAccessPolicy
+from api.repositories.jobs import JobsRepository, JobFilters
+from api.repositories.providers import ProviderRepository
+from api.repositories.functions import FunctionRepository
+
+logger = logging.getLogger("gateway.use_cases.get_jobs")
+
+
+class ProviderNotFoundException(Exception):
+    """Provider not found or access denied."""
+
+
+class FunctionNotFoundException(Exception):
+    """Function not found or access denied."""
+
```

Comment:
In the files migration i created a generic `NotFoundException` in the domain folder: https://github.com/Qiskit/qiskit-serverless/pull/1696/files#diff-e6da6866795b4c157d9b9e887eb27622f036a9dc512d30e5097a301df1d5001c

Also I created an additional decorator to quickly manage the possible `NotFoundException`s (and other that can be created later) and create the corresponding `Response`. The only cons is that you have to set the message when you create the exception to inform about the issue.